stackslib/src/net/api/getattachmentsinv.rs

```diff
@@ -0,0 +1,269 @@
+            return StacksHttpResponse::new_error(&preamble, &HttpBadRequest::new(msg))
+                .try_into_contents()
+                .map_err(NetError::from);
+        }
+        if page_indexes.len() == 0 {
+            let msg = format!("Page indexes missing");
+            warn!("{}", msg);
+            return StacksHttpResponse::new_error(&preamble, &HttpBadRequest::new(msg))
+                .try_into_contents()
+                .map_err(NetError::from);
+        }
+
+        let mut pages = vec![];
+
+        for page_index in page_indexes.iter() {
+            let page_res =
+                node.with_node_state(|network, _sortdb, _chainstate, _mempool, _rpc_args| {
+                    match network
+                        .get_atlasdb()
+                        .get_attachments_available_at_page_index(*page_index, &index_block_hash)
+                    {
+                        Ok(inventory) => Ok(AttachmentPage {
+                            inventory,
+                            index: *page_index,
+                        }),
+                        Err(e) => {
+                            let msg = format!("Unable to read Atlas DB - {}", e);
+                            warn!("{}", msg);
+                            Err(msg)
+                        }
+                    }
+                });
+
+            match page_res {
+                Ok(page) => {
+                    pages.push(page);
+                }
+                Err(msg) => {
+                    return StacksHttpResponse::new_error(&preamble, &HttpNotFound::new(msg))
+                        .try_into_contents()
+                        .map_err(NetError::from);
+                }
+            }
+        }
+
+        let content = GetAttachmentsInvResponse {
+            block_id: index_block_hash.clone(),
+            pages,
+        };
+
+        let mut preamble = HttpResponsePreamble::ok_json(&preamble);
+        preamble.set_canonical_stacks_tip_height(Some(node.canonical_stacks_tip_height()));
+        let body = HttpResponseContents::try_from_json(&content)?;
+        Ok((preamble, body))
+    }
+}
+
+/// Decode the HTTP response
+impl HttpResponse for RPCGetAttachmentsInvRequestHandler {
+    fn try_parse_response(
+        &self,
+        preamble: &HttpResponsePreamble,
+        body: &[u8],
+    ) -> Result<HttpResponsePayload, Error> {
+        let pages: GetAttachmentsInvResponse = parse_json(preamble, body)?;
+        Ok(HttpResponsePayload::try_from_json(pages)?)
+    }
+}
+
+impl StacksHttpRequest {
+    /// Make a new request for attachment inventory page
+    pub fn new_getattachmentsinv(
+        host: PeerHost,
+        index_block_hash: StacksBlockId,
+        page_indexes: HashSet<u32>,
+    ) -> StacksHttpRequest {
+        let page_list: Vec<String> = page_indexes.into_iter().map(|i| format!("{}", i)).collect();
+        StacksHttpRequest::new_for_peer(
+            host,
+            "GET".into(),
```

Comment:
We should use the self.verb() function I think instead of the hardcoded "GET" string for maintainability.

Reply:
We can't -- `StacksHttpRequest` does not have `self.verb()`.  The `RPCGetAttachmentsInvRequestHandler` struct does, as part of implementing `HttpRequestHandler` (and `RPCRequestHandler`).  However, this has to be an instance method so the correct function body can get called.  This is required in `StacksHttp::try_handle_request()`, which calls the `.verb()` and `.regex()` functions on a `Box<dyn RPCRequestHandler>` in order to dispatch to the correct implementations' methods.  If we didn't do this, then we'd need to somehow identify the trait's implementation type at runtime to handle the dispatch.